src/transformers/models/blip/modeling_tf_blip_text.py

```diff
@@ -1060,7 +1060,8 @@ def call(
             labels = labels[:, 1:]
             labels = tf.reshape(labels, (-1,))
             # Keras won't give us label smoothing for sparse CE, so we de-sparsify things here
-            one_hot_labels = tf.one_hot(labels, depth=self.config.vocab_size, dtype=tf.float32)
+            # Use relu to clamp masked labels at 0 to avoid NaN (we will be zeroing those out later anyway)
+            one_hot_labels = tf.one_hot(tf.nn.relu(labels), depth=self.config.vocab_size, dtype=tf.float32)
```

Comment:
Yes, but the masked positions are set to 0 by the line `lm_loss *= masked_positions`! However, if we get a NaN value because we passed a negative label to an internal TF function, that won't be zeroed out correctly.

Reply:
OK - works for me!